doc/psud/PSU_daemon_design.md

```diff
@@ -0,0 +1,152 @@
+# SONiC PSU Daemon Design #
+
+### Rev 0.1 ###
+
+### Revision ###
+
+ | Rev |     Date    |       Author       | Change Description                |
+ |:---:|:-----------:|:------------------:|-----------------------------------|
+ | 0.1 |             |      Chen Junchao  | Initial version                   |
+
+
+## 1. Overview
+
+When I write this document, SONiC already has a PSU daemon called psud running in pmon docker. There is already a high level design document which contains existing psud [design](https://github.com/Azure/SONiC/blob/master/doc/pmon/pmon-enhancement-design.md), but some of its content is out of date. This document will update the design of current PSU daemon implementation and add an enhancement about PSU led management.
+
+The purpose of PSU daemon is to collect platform PSU data and trigger proper actions if necessary. Major functions of psud include:
+
+- Collect constant PSU data during daemon boot up, such as PSU number.
+- Collect variable PSU data periodically.
+- Monitor PSU event, set LED color and trigger syslog according to event type.
+
+## 2. PSU data collection
+
+PSU daemon data collection flow diagram:
+
+![](https://github.com/keboliu/SONiC/blob/master/doc/pmon/daemon-flow.svg)
```

Comment:
we need to upload this flow chart to community repo, seems still in my personal fork.